scripts/generateCliCode.ts

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright © 2022 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * This script generates the options module used for the `set` and `start` subcommands,
+ * according to the preferences spec from pkg/rancher-desktop/assets/specs/command-api.yaml
+ */
+
+'use strict';
+
+import fs from 'fs';
+import path from 'path';
+
+import ejs from 'ejs';
+import yaml from 'yaml';
+
+interface commandType {
+    flagType: string;
+    flagName: string;
+    flagOption: string;
+    defaultValue: string;
+    usageNote: string;
+    miscNote: string;
+    aliasFor: string;
+    enums: string[] | undefined;
+}
+
+interface commandLineOptionFields {
+  propertyName: string;
+  capitalizedName: string;
+  lcTypeName: string;
+  aliasFor: string;
+  enums: string[] | undefined;
+}
+
+type yamlObject = any; // Record<string, string|boolean|number|Record<string, any>>;
```

Comment:
Don't need the commented out part. (I assume it's there due to issues with trying to define a type that references itself.)